commons/src/main/java/io/aiven/kafka/connect/common/source/input/Transformer.java

```diff
@@ -24,21 +24,25 @@
 import java.util.stream.Stream;
 import java.util.stream.StreamSupport;
 
-import org.apache.kafka.common.config.AbstractConfig;
 import org.apache.kafka.connect.data.SchemaAndValue;
 
+import io.aiven.kafka.connect.common.config.SourceCommonConfig;
+
 import org.apache.commons.io.function.IOSupplier;
 import org.slf4j.Logger;
 
 public abstract class Transformer {
 
-    public abstract void configureValueConverter(Map<String, String> config, AbstractConfig sourceConfig);
+    public final static long UNKNOWN_STREAM_LENGTH = -1;
+
+    public abstract void configureValueConverter(Map<String, String> config, SourceCommonConfig sourceConfig);
 
     public final Stream<SchemaAndValue> getRecords(final IOSupplier<InputStream> inputStreamIOSupplier,
-            final String topic, final int topicPartition, final AbstractConfig sourceConfig, final long skipRecords) {
+            final long streamLength, final String topic, final int topicPartition,
+            final SourceCommonConfig sourceConfig, final long skipRecords) {
 
-        final StreamSpliterator spliterator = createSpliterator(inputStreamIOSupplier, topic, topicPartition,
-                sourceConfig);
+        final StreamSpliterator spliterator = createSpliterator(inputStreamIOSupplier, streamLength, topic,
```

Comment:
```suggestion
        final StreamSpliterator spliterator = createSpliterator(inputStreamIOSupplier, objectSize, topic,
```
Stream does not have a length in general. Here iterator is sending object size, and in this repo, we are dealing with objects.

Reply:
It is referring to the stream which has been input into the createSpliterator method.
This is commons code and is not specific to the object size but the stream length.
changing the name would be a misnomer i feel.

Reply:
This repo mainly deals with cloud storage objects. Even if it's common, and when we use it for azure or gcp, I think objectSize or something similar looks better to me

Reply:
I'll let @Claudenw be the deciding vote on that one 👍